assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1514,6 +1514,23 @@ protected String build_student_view_submission_context(VelocityPortlet portlet,
 				context.put("student",student);
 			}
 		}
+		
+		//TODO move upwards inside some checking
+		Site site = null;
+		try {
+			site = SiteService.getSite(contextString);
```

Comment:
Loading the site doesn't look to be the pattern `buildMainPanelContext` puts the site in the context for most templates and it should probably so it for this one.